helix-core/src/main/java/org/apache/helix/spectator/RoutingTableSnapshot.java

```diff
@@ -145,4 +152,34 @@ public RoutingTableSnapshot(RoutingTable routingTable) {
   public Collection<ExternalView> getExternalViews() {
     return _routingTable.getExternalViews();
   }
+
+  /**
+   * Returns a Collection of latest snapshot of CustomizedView. Note that if the RoutingTable is
+   * instantiated using CurrentStates, this Collection will be empty.
+   * @return
+   */
+  public Collection<CustomizedView> getCustomizeViews() {
+    if (_propertyType.equals(PropertyType.CUSTOMIZEDVIEW)){
+    CustomizedViewRoutingTable customizedViewRoutingTable =
+        (CustomizedViewRoutingTable) _routingTable;
+    return customizedViewRoutingTable.geCustomizedViews();
+    }
+    return Collections.emptySet();
+  }
+
+  /**
+   * Returns the PropertyType associated with this RoutingTableSnapshot
+   * @return
+   */
+  public PropertyType getPropertyType() {
+    return _propertyType;
+  }
+
+  /**
+   * Return the Type associated with thi RoutingTableSnapshot (mainly used for CustomizedView)
+   * @return
+   */
+  public String getCustomizedStateType() {
```

Comment:
getStateType() ?

Reply:
getCustomizedStateType would be clearer. As this is a snapshot class for different kinds of views.

Reply:
I don't have preference. I keep it for now as it is. We can change it later.